### PR TITLE
Add API timings to observers

### PR DESCRIPTION
**Pull request by ausmith**
Lay the groundwork for starting to collect API call timings via our existing observers like https://github.com/cbdr/consumer-main/blob/master/config/initializers/cb_api.rb#L26-L48 and https://github.com/cbdr/consumer-domain/blob/master/lib/rails/cb_api_observer.rb

### CODE REVIEW

**Comment by ausmith** [comment]
Making an adjustment, I think I'm pulling the timings at the wrong time.

**Comment by ausmith** [comment]
I have an idea for how to make this a non-breaking change, going to switch to that.

**Comment by ausmith** [comment]
Let's try without breaking changes!

File: lib/cb/utils/api.rb [hunk 1]
```diff
@@ -46,9 +46,10 @@ def cb_delete(path, options={}, &block)
 
       def execute_http_request(http_method, uri, path, options={}, &block)
         self.class.base_uri(uri || Cb.configuration.base_uri)
-        cb_event(:"cb_#{http_method.to_s}_before", path, options, nil, &block)
+        start_time = Time.now.to_f
+        cb_event(:"cb_#{http_method.to_s}_before", path, options, nil, start_time - start_time, &block)
         response = self.class.method(http_method).call(path, options)
-        cb_event(:"cb_#{http_method.to_s}_after", path, options, response, &block)
+        cb_event(:"cb_#{http_method.to_s}_after", path, options, response, Time.now.to_f - start_time, &block)
```
Comment (helious) [inline lib/cb/utils/api.rb]: you get the `.to_s` for free from the interp, btw! (i know it was there before)

Comment (ausmith) [inline lib/cb/utils/api.rb]: Agreed, I didn't even notice that that was there before :smile: Boy scouted some other portions of this though. Lemme get rid of that `.to_s` real quick.

**Comment by brentfisher** [comment]
this is awesome +1

File: spec/support/mocks/observer.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module Mocks
   class Observer
     def update(api_call)
-      puts "hello I am observer for #{api_call.api_call_type.to_s} #{api_call.path}"
+      # puts "hello I am observer for #{api_call.api_call_type.to_s} #{api_call.path}"
```
Comment (tkling) [inline spec/support/mocks/observer.rb]: Such comment O__O

File: lib/cb/utils/api.rb [hunk 3]
```diff
@@ -46,9 +46,10 @@ def cb_delete(path, options={}, &block)
 
       def execute_http_request(http_method, uri, path, options={}, &block)
         self.class.base_uri(uri || Cb.configuration.base_uri)
-        cb_event(:"cb_#{http_method.to_s}_before", path, options, nil, &block)
+        start_time = Time.now.to_f
+        cb_event(:"cb_#{ http_method }_before", path, options, nil, start_time - start_time, &block)
```
Comment (nivardus) [inline lib/cb/utils/api.rb]: `start_time - start_time` ?

Comment (ausmith) [inline lib/cb/utils/api.rb]: So we get a `0` before the event has happened and a real number afterward.

Comment (ausmith) [inline lib/cb/utils/api.rb]: `nil` just didn't make sense for that field where as indicating the amount of time that has elapsed so far did.

Comment (nivardus) [inline lib/cb/utils/api.rb]: I would think `0` might be less confusing than doing the arithmetic to get 0 here.

Comment (ausmith) [inline lib/cb/utils/api.rb]: Yeah, was thinking that as well as I typed that response out. One sec, will do real quick.

**Comment by nivardus** [comment]
This PR will be damn helpful.

**Comment by helious** [comment]
i likes it :+1:

**Comment by ausmith** [comment]
I see a whole lot of :+1:s but not a lot of merge button clicking! :smiley:

**Pull request merged**